### ISSUE TITLE
#[Object] macro panic with default u32

### ISSUE THREAD

**Issue by stevenxie** [async-graphql/async-graphql#361]
When defining `#[Object]` resolvers, this code compiles:
```rust
    /// List all registered `Shelter`s.
    async fn shelters(
        &self,
        ctx: &Context<'_>,

        #[graphql(
            desc = "The maximum number of `Shelter`s to fetch.",
            default = 25
        )]
        limit: i32,

        #[graphql(desc = "The number of initial `Shelter`s to skip.", default)]
        offset: u32,
    ) -> FieldResult<Vec<Shelter>> {
     todo!()
}
```

However, this fails to compile:
```rust
    /// List all registered `Shelter`s.
    async fn shelters(
        &self,
        ctx: &Context<'_>,

        #[graphql(
            desc = "The maximum number of `Shelter`s to fetch.",
            default = 25
        )]
        limit: u32,

        #[graphql(desc = "The number of initial `Shelter`s to skip.", default)]
        offset: u32,
    ) -> FieldResult<Vec<Shelter>> {
      todo!()
}
```

Error:
```
error[E0308]: mismatched types
  --> src/graphql/query.rs:17:1
   |
17 | #[Object]
   | ^^^^^^^^^ expected `u32`, found `i32`
   |
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
help: you can convert an `i32` to `u32` and panic if the converted value wouldn't fit
   |
17 | (#[Object]).try_into().unwrap()
   |
```

My current workaround is to use `graphql(default_with = "25")` for an unsigned input number. However, I believe it should be possible to use `graphql(default = 25)` for both unsigned and signed input numbers.

Note that the same error occurs if I specify the type on the default value, i.e. `graphql(default = 25u32)`.

## Specifications

- Version: `async-graphql` 2.4.3, `rustc` 1.48.0 (7eac88abb 2020-11-16) [stable, x86-via-rosetta]
- Platform: MacOS 2020 M1

**Comment by sunli829** [comment]
Thank you for your suggestion, and I will review the code implementation here.

**Comment by sunli829** [comment]
Released `v2.4.3`. 🙂
